File: src/main/java/pl/training/workshop/m8/s12_expandcontract/Booking.java
package pl.training.workshop.m8.s12_expandcontract;

import java.util.List;

import pl.training.workshop.shared.Money;

/** Stabilny kontrakt sceny: rezerwacja zapisywana w bazie. */
public record Booking(String id, String email, List<String> seats, Money total) {
    public Booking {
        seats = List.copyOf(seats);
    }
}

File: csharp/src/Training.Workshop/M8/S12ExpandContract/Booking.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M8.S12ExpandContract;

/// <summary>
/// Stabilny kontrakt sceny: rezerwacja zapisywana w bazie. Równość po wartości - także listy
/// miejsc (rekord C# porównywałby listę po referencji).
/// </summary>
public sealed record Booking(string Id, string Email, IReadOnlyList<string> Seats, Money Total)
{
    public IReadOnlyList<string> Seats { get; } = Seats.ToList();

    public bool Equals(Booking? other)
    {
        return other is not null
            && Id == other.Id
            && Email == other.Email
            && Seats.SequenceEqual(other.Seats)
            && Total.Equals(other.Total);
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Id, Email, string.Join(",", Seats), Total);
    }

    public override string ToString()
    {
        return "Booking[" + Id + ", " + Email + ", " + string.Join(",", Seats) + ", " + Total + "]";
    }
}
